server/blob/blob-s3/src/test/java/org/apache/james/blob/objectstorage/aws/BucketNameResolverTest.java

```diff
@@ -50,10 +65,45 @@ void resolveShouldReturnValueWhenNamespace() {
             assertThat(resolver.resolve(BucketName.of("namespace")))
                 .isEqualTo(BucketName.of("namespace"));
         }
+
+        @Test
+        void unresolveShouldReturnPassedValue() {
```

Comment:
Isn't that case checked above in the parameterized test `withShouldAddNewValuesInSet`?

Reply:
In a more general manner, isn't this the same for tests in other nested classes?

Reply:
> In a more general manner, isn't this the same for tests in other nested classes?

Yes but un top of different bucket resolver configuration and it (might or might not) yield different results.

Being very sure of the behaviour of that class is very important to me, I assume 100% having a very detailed test suite here.

Reply:
Ok got your point

Reply:
For the mathematical background https://en.wikipedia.org/wiki/Inverse_function <3